torch/__init__.py

```diff
@@ -507,85 +537,176 @@ def __rpow__(self, other):
         else:
             return sym_float(self).__rpow__(sym_float(other))
 
-    def __eq__(self, other: object) -> builtins.bool:
+    def __eq__(self, other: _Any) -> "_bool":
```

Comment:
`SymInt` does not actually support equality comparison against arbitrary Python objects (as is usually the case for `__eq__`). Changed to `_Any` as it didn't seem that important to me to determine the set of supported objects...